testutil/testmigration/fixtures_recoverable_account.go

```diff
@@ -0,0 +1,646 @@
+	return func(config *MorseFixturesConfig) {
+		config.OrphanedActorsConfig = cfg
+	}
+}
+
+// WithUnstakedAccountBalancesFn sets the UnstakedAccountBalancesConfig for the fixtures.
+// It defines how balances are determined for unstaked accounts.
+func WithUnstakedAccountBalancesFn(cfg UnstakedAccountBalancesConfigFn) MorseFixturesOption {
+	return func(config *MorseFixturesConfig) {
+		config.UnstakedAccountBalancesConfigFn = cfg
+	}
+}
+
+// WithValidatorStakesFn sets the ValidatorStakesConfig for the fixtures.
+// It defines how staked and unstaked balances are determined for validator accounts.
+func WithValidatorStakesFn(cfg ValidatorStakesConfigFn) MorseFixturesOption {
+	return func(config *MorseFixturesConfig) {
+		config.ValidatorStakesConfigFn = cfg
+	}
+}
+
+// WithApplicationStakesFn sets the ApplicationStakesConfig for the fixtures.
+// It defines how staked and unstaked balances are determined for application accounts.
+func WithApplicationStakesFn(cfg ApplicationStakesConfigFn) MorseFixturesOption {
+	return func(config *MorseFixturesConfig) {
+		config.ApplicationStakesConfigFn = cfg
+	}
+}
+
+// NewMorseFixtures creates a new MorseMigrationFixtures instance with the provided options.
+// It initializes the necessary data structures and generates test fixtures according to
+// the configuration provided through the options.
+func NewMorseFixtures(opts ...MorseFixturesOption) (*MorseMigrationFixtures, error) {
+	morseFixtures := &MorseMigrationFixtures{
+		morseStateExport: &migrationtypes.MorseStateExport{
+			AppState: &migrationtypes.MorseTendermintAppState{
+				Application: &migrationtypes.MorseApplications{
+					Applications: make([]*migrationtypes.MorseApplication, 0),
+				},
+				Auth: &migrationtypes.MorseAuth{
+					Accounts: make([]*migrationtypes.MorseAuthAccount, 0),
+				},
+				Pos: &migrationtypes.MorsePos{
+					Validators: make([]*migrationtypes.MorseValidator, 0),
+				},
+			},
+		},
+		morseAccountState: &migrationtypes.MorseAccountState{
+			Accounts: make([]*migrationtypes.MorseClaimableAccount, 0),
+		},
+	}
+
+	// Apply all the provided functional options to configure the fixtures
+	morseFixtures.config = &MorseFixturesConfig{}
+	for _, opt := range opts {
+		opt(morseFixtures.config)
+	}
+
+	totalAccounts := morseFixtures.config.ValidAccountsConfig.NumAccounts +
+		morseFixtures.config.ValidAccountsConfig.NumApplications +
+		morseFixtures.config.ValidAccountsConfig.NumValidators +
+		morseFixtures.config.ValidAccountsConfig.NumModuleAccounts +
+		morseFixtures.config.InvalidAccountsConfig.NumAddressTooShort +
+		morseFixtures.config.InvalidAccountsConfig.NumAddressTooLong +
+		morseFixtures.config.InvalidAccountsConfig.NumNonHexAddress +
+		morseFixtures.config.OrphanedActorsConfig.NumApplications +
+		morseFixtures.config.OrphanedActorsConfig.NumValidators
+
+	morseFixtures.morseAccountState.Accounts = make([]*migrationtypes.MorseClaimableAccount, totalAccounts)
```

Comment:
This means that we're ALWAYS accessing by index and NEVER appending, right?

Reply:
Yes,